### DISCUSSION TITLE
Using in-line chunks within figure/table captions

### DISCUSSION THREAD

**Discussion by LuisLauM** [quarto-dev/quarto-cli#9672] [Q&A] [answered]
### Description

Is it possible to use in-line chunks inside the caption of a figure (or table)? I know there is a way to do it using paste functions (`paste`/`sprintf`), but I would like to know if Quarto has implemented a way to use the standard method, for instance: 'The sum of mpg values is `r sum(mtcars$mpg)`'.

By the way, apparently the use of `sprintf` is not supported when used inside a `#|`:
```
---
title: "test"
format: docx
---

## Quarto

Testing ref: @fig-1 and @fig-2.

```{r fig.cap=sprintf("hola %s", "mundo")}
#| label: fig-1
#| echo: false

plot(mtcars$mpg, mtcars$disp)
```

```{r}
#| label: fig-2
#| echo: false
#| fig-cap: sprintf("hola %s", "mundo")

plot(mtcars$disp, mtcars$mpg)
```

```

![image](https://github.com/quarto-dev/quarto-cli/assets/8452334/c064ca32-73ca-4564-95db-aaec8c53c793)

**Comment by mcanouil** [comment, marked as answer]
Could you edit the title of your post with a meaningful one? Currently, it's the placeholder.
Could you also properly format your post using code blocks for code and terminal outputs?
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.

Thank you!


Regarding the question, see <https://quarto.org/docs/computations/r.html#chunk-options>.
Not that this is not a Quarto feature, it's `knitr`, thus it won't work for other engines.
